src/libs/actions/Task.js

```diff
@@ -0,0 +1,209 @@
+        optimisticAssigneeAddComment.reportAction.message[0].taskReportID = optimisticTaskReport.reportID;
+    }
+
+    // Create the CreatedReportAction on the task
+    const optimisticTaskCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(optimisticTaskReport.reportID);
+
+    const optimisticAddCommentReport = ReportUtils.buildOptimisticAddCommentReportAction(`[Created a task](tbd/r/${optimisticTaskReport.reportID}): ${title}`);
+    optimisticAddCommentReport.reportAction.message[0].taskReportID = optimisticTaskReport.reportID;
+
+    const optimisticData = [
+        {
+            onyxMethod: Onyx.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${optimisticTaskReport.reportID}`,
+            value: optimisticTaskReport,
+        },
+        {
+            onyxMethod: Onyx.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${optimisticTaskReport.reportID}`,
+            value: {[optimisticTaskCreatedAction.reportActionID]: optimisticTaskCreatedAction},
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${parentReportID}`,
+            value: optimisticAddCommentReport,
+        },
+    ];
+
+    if (optimisticAssigneeAddComment) {
+        optimisticData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${assigneeChatReportID}`,
+            value: optimisticAssigneeAddComment,
+        });
+    }
+
+    const successData = [];
+
+    const failureData = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${optimisticTaskReport.reportID}`,
+            value: null,
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${optimisticTaskReport.reportID}`,
+            value: {[optimisticTaskCreatedAction.reportActionID]: {pendingAction: null}},
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${parentReportID}`,
+            value: {[optimisticAddCommentReport.reportActionID]: {pendingAction: null}},
+        },
+    ];
+
+    if (optimisticAssigneeAddComment) {
+        failureData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${assigneeChatReportID}`,
+            value: {[optimisticAssigneeAddComment.reportActionID]: {pendingAction: null}},
+        });
+    }
+
+    API.write(
+        'CreateTask',
+        {
+            parentReportActionID: optimisticAddCommentReport.reportActionID,
+            parentReportID,
+            taskReportID: optimisticTaskReport.taskReportID,
+            reportName: optimisticTaskReport.reportName,
+            title: optimisticTaskReport.reportName,
+            description: optimisticTaskReport.description,
+            assignee,
+            assigneeChatReportID,
```

Comment:
We're still missing a couple more parameters to go to the API

- parentReportActionID (the addComment reportActionID on the parent report)
- createdTaskReportActionID (the CREATED reportActionID on the parent report)
- assigneeChatReportActionID (when the assignee is assigned and you need to create an addComment reportAction to the assignee chatReportID)

Reply:
ah looks like you are passing the parentReportActionID already, but it's showing as empty (left a comment on it)

Reply:
Note: This is weird cause the createdActions and AddComment actions are slightly different

For the createdReportAction you can just do optimisticTaskCreatedAction.reportActionID but for the addComment reportAction you have to do optimisticAddCommentReport.reportAction.reportActionID

Reply:
I believe this has been resolved!